internal/dag/httpproxy_processor.go

```diff
@@ -640,6 +653,11 @@ func (p *HTTPProxyProcessor) computeRoutes(
 				r.Clusters = append(r.Clusters, c)
 			}
 		}
+		if len(r.Clusters) == 0 {
+			r.DirectResponse = &DirectResponse{
```

Comment:
Might be helpful now that this is used in multiple places to centralize the logic so that if it changes, we'll have a single place to update.

Reply:
I may have misunderstood so please correct me! I wonder wouldn't the logic around direct responses always be different, making it hard to centralize? Now currently there are two direct responses by this PR:

* Here in this spot we have processed whole `routes`-section without failures, including missing service which is now passed through, but we still have no clusters to route to -> instantiate `StatusServiceUnavailable`
* Another case is in when `includes`-section is processed, we did not find target of include, but we have known condition -> instantiate `StatusBadGateway`

So will the logic always depend on the "context"?  When we cover more error scenarios from the design doc, wouldn't that logic also need to be sprinkled around each error handling branch, not leaving any logic to be centralized?

Reply:
I was more suggesting that the `&DirectResponse` bit might be hidden inside a method somewhere and the specifics could be passed into that method. This way if the struct changed in Envoy, we'd update it in one place not multiple.

Reply:
Fixed.